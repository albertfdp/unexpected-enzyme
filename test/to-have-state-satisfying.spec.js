import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React, { Component } from 'react';

class Counter extends Component {
  constructor(props) {
    super(props);

    this.increment = this.increment.bind(this);

    this.state = {
      count: this.props.count
    };
  }

  increment() {
    this.setState(prev => ({
      count: prev.count + 1
    }));
  }

  render() {
    const { count } = this.state;

    return (
      <div className="count">
        {count}
        <button className="increment" onClick={this.increment}>
          Increment
        </button>
      </div>
    );
  }
}

describe('to-have-state-satisfying', () => {
  describe('<assertion>', () => {
    describe('to-have-state', () => {
      it('passes when the actual matches the expected', () => {
        const wrapper = mount(<Counter count={0} />);

        expect(wrapper, 'to have state satisfying to have keys', ['count']);

        wrapper.find('button').simulate('click');

        expect(wrapper, 'to have state satisfying to have keys', ['count']);
      });

      it('fails when the actual does not match the expected', () => {
        expect(
          () =>
            expect(
              mount(<Counter count={0} />),
              'to have state satisfying to have keys',
              ['caunt', 'number']
            ),
          'with error matching snapshot'
        );
      });

      describe('when negated', () => {
        it('passes when the actual does not match the expected', () => {
          expect(
            mount(<Counter count={0} />),
            'to have state satisfying not to have keys',
            ['foobar']
          );
        });

        it('fails when the actual does match the expected', () => {
          expect(
            () =>
              expect(
                mount(<Counter count={0} />),
                'to have state satisfying not to have keys',
                ['count']
              ),
            'with error matching snapshot'
          );
        });
      });
    });
  });

  describe('<object>', () => {
    it('passes when the passed matches the expected', () => {
      const wrapper = mount(<Counter count={0} />);

      expect(wrapper, 'to have state satisfying', {
        count: 0
      });

      wrapper.find('button').simulate('click');

      expect(wrapper, 'to have state satisfying', {
        count: 1
      });
    });

    it('fails when the actual does not match the expected', () => {
      expect(
        () =>
          expect(mount(<Counter count={0} />), 'to have state satisfying', {
            count: 3,
            number: 0
          }),
        'with error matching snapshot'
      );
    });
  });
});
