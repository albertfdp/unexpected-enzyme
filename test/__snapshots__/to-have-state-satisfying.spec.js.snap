// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-have-state-satisfying <assertion> fails when the actual does not match the expected 1`] = `
"expected <div class=\\"count\\">0<button class=\\"increment\\">Increment</button></div>
to have state satisfying to have keys [ 'caunt', 'number' ]"
`;

exports[`to-have-state-satisfying <assertion> when negated fails when the actual does match the expected 1`] = `
"expected <div class=\\"count\\">0<button class=\\"increment\\">Increment</button></div>
to have state satisfying not to have keys [ 'count' ]"
`;

exports[`to-have-state-satisfying <object> fails when the actual does not match the expected 1`] = `
"expected <div class=\\"count\\">0<button class=\\"increment\\">Increment</button></div>
to have state satisfying { count: 3, number: 0 }

{
  count: 0 // should equal 3
  // missing number: 0
}"
`;
